test/e2e/kafka_source_test.go

```diff
@@ -42,26 +45,138 @@ import (
 )
 
 const (
-	kafkaBootstrapUrl     = "my-cluster-kafka-bootstrap.kafka.svc:9092"
-	kafkaClusterName      = "my-cluster"
-	kafkaClusterNamespace = "kafka"
+	kafkaBootstrapUrlPlain   = "my-cluster-kafka-bootstrap.kafka.svc:9092"
+	kafkaBootstrapUrlTLS     = "my-cluster-kafka-bootstrap.kafka.svc:9093"
+	kafkaBootstrapURLSASL    = "my-cluster-kafka-bootstrap.kafka.svc:9094"
+	kafkaClusterName         = "my-cluster"
+	kafkaClusterNamespace    = "kafka"
+	kafkaClusterSASLUsername = "my-sasl-user"
+	kafkaClusterTLSUsername  = "my-tls-user"
+	kafkaClusterPassword     = "my-seekrit-password"
```

Comment:
In the `consolidated-sasl` test run, we create a secrte before the actual run:
https://github.com/knative-sandbox/eventing-kafka/blob/master/test/e2e-tests.sh#L494

And the secret gets all the required settings for SASL/TLS, from Strimzi (CA CERT and the SASL USER):
https://github.com/knative-sandbox/eventing-kafka/blob/master/test/e2e-tests.sh#L369-L372

Reply:
How about:

1) we create a new `kafka_source_sasl_test.go` file, 
https://github.com/knative-sandbox/eventing-kafka/blob/192986e75e1db91bf45fd9dd45fba9176ce9bfe4/test/e2e/kafka_source_test.go#L1-L2

2) add a `sasl` build-tag after https://github.com/knative-sandbox/eventing-kafka/blob/192986e75e1db91bf45fd9dd45fba9176ce9bfe4/test/e2e/kafka_source_test.go#L1-L2 

e.g.:
```
//+build sasl
```

3) Run that test only on the SASL test run:
https://github.com/knative-sandbox/eventing-kafka/blob/master/test/e2e-tests.sh#L412
e.g. change that line to:

```
...
  go_test_e2e -tags=e2e,sasl -timeout=40m -test.parallel=${TEST_PARALLEL} ./test/e2e -channels=messaging.knative.dev/v1beta1:KafkaChannel  || fail_test

```

Reply:
I added TLS support for the tests before seeing this comment.  Because there are already both -- I just opted to move the auth creation before the channel tests and they all pass.  Please let me know if you'd prefer that reworked (but figure, a couple extra secrets isn't the worst tradeoff? I could be wrong :)